src/Console/Commands/EmitEvents.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Softonic\TransactionalEventPublisher\Console\Commands;
+
+use Exception;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\LazyCollection;
+use RuntimeException;
+use Softonic\TransactionalEventPublisher\Contracts\EventStoreMiddlewareContract;
+use Softonic\TransactionalEventPublisher\Model\DomainEvent;
+use Softonic\TransactionalEventPublisher\Model\DomainEventsCursor;
+
+class EmitEvents extends Command
+{
+    /**
+     * Max delay for errors retry in seconds (1 minute).
+     */
+    private const MAX_DELAY_FOR_ERRORS = 60;
+
+    /**
+     * Max delay for no events retry in microseconds (1 second).
+     */
+    private const MAX_DELAY_FOR_NO_EVENTS = 1_000_000;
+
+    /**
+     * Base delay for no events retry in microseconds (1 millisecond).
+     */
+    private const BASE_DELAY_FOR_NO_EVENTS = 1000;
+
+    protected $signature = 'event-sourcing:emit
+        {--dbConnection=mysql : Indicate the database connection to use (MySQL unbuffered for better performance when large amount of events)}
+        {--batchSize=100 : Indicate the amount of events to be sent per publish. Increase for higher throughput}
+        {--allEvents : Option to send all the events from the beginning by resetting the cursor}';
+
+    protected $description = 'Continuously emits domain events in batches';
+
+    public EventStoreMiddlewareContract $eventPublisherMiddleware;
```

Comment:
Why are all these properties public?

Reply:
For the test... 😞 In the test cases that I test the public method sendBatch() to avoid the infinite loop, I need to set some properties that have not been initialised, which is done in the handle() method.

Reply:
I think you can use mockery for that, but ok